dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/task/AbstractJdbcTask.java

```diff
@@ -82,7 +93,28 @@ protected CSVFormat newCsvFormat(@Nonnull ResultSet rs) throws SQLException {
     Class<? extends Enum<?>> headerClass = getHeaderClass();
     if (headerClass != null) {
       format = format.withHeader(headerClass);
-      if (headerClass.getEnumConstants().length != rs.getMetaData().getColumnCount())
+      ResultSetMetaData metaData = rs.getMetaData();
+      if (isAssessment) {
+
+        HashSet<String> columnNames = new HashSet<>();
+
+        for (int i = 0; i < metaData.getColumnCount(); i++) {
+          columnNames.add(metaData.getColumnName(i));
+        }
+
+        String missingColumns =
+            Arrays.stream(headerClass.getEnumConstants())
+                .map(Enum::name)
+                .map(name -> CaseFormat.UPPER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, name))
+                .filter(columnName -> !columnNames.contains(columnName))
```

Comment:
optional: this could alternatively be done by using a https://docs.oracle.com/javase/8/docs/api/java/util/Set.html#removeAll

Reply:
I didn't want to add everything to a set and then remove unwanted ones and then `stream` again for joining. wdyt?

Reply:
I don't see the need to join again, because the column difference is just informational, and you can always copy the set. Set operations is usually faster than cross-iterating through both iterators.

IIUC, in current implementation this check is executed for every table-interval, so for a 30d total duration it would be around 24/day-table*30d*1table + 1/day-table*30d*15table = 1170 operations and not increasing terribly fast, so it is not a key point of optimization, and this change is optional.

One definitely nice optimization is to only execute it once for every table, though.